### ISSUE TITLE
ImportError: cannot import name 'rng_integers' from 'scipy._lib._util'

### ISSUE THREAD

**Issue by ZihanChen1995** [stellargraph/stellargraph#1784]
### Describe the bug

Hi! Thank you for this great library. I'm trying to install Stellargraph on Ubuntu 18.04. However, after I successfully installed Stellargraph via conda ([follow this instruction](https://stellargraph.readthedocs.io/en/stable/README.html#installation)), when I try to import Stellargraph, this problem happened:

```
ImportError: cannot import name 'rng_integers' from 'scipy._lib._util' (/home/zihan/anaconda3/envs/TF2/lib/python3.7/site-packages/scipy/_lib/_util.py)
```

To avoid the possible problem coming from `scipy`, I reinstalled it, and made sure it successfully installed in this environment:

```
conda list -n TF2

scipy                     1.4.1                    pypi_0    pypi
stellargraph              1.2.1                      py_0    stellargraph
tensorflow                2.2.0           mkl_py37h6e9ce2d_0  
```

Hence I'm quite confused with why this problem happens. Could you please help me with this issue? Thank you a lot in advance!

### Environment

Operating system: Ubuntu 18.04

Python version: 3.7.7

Package versions: stellargraph==1.2.1, tensorflow==2.2.0

***

More detailed error description:
```
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-11-855dad28d1e4> in <module>
----> 1 import stellargraph as sg
      2 #from stellargraph.data import EdgeSplitter
      3 #from stellargraph.mapper import FullBatchLinkGenerator
      4 #from stellargraph.layer import GCN, LinkEmbedding
      5 # import scipy

~/anaconda3/envs/TF2/lib/python3.7/site-packages/stellargraph/__init__.py in <module>
     37 
     38 # Import modules
---> 39 from stellargraph import (
     40     data,
     41     calibration,

~/anaconda3/envs/TF2/lib/python3.7/site-packages/stellargraph/data/__init__.py in <module>
     21 
     22 # Expose the stellargraph.data classes:
---> 23 from .explorer import *
     24 from .edge_splitter import *
     25 from .node_splitter import *

~/anaconda3/envs/TF2/lib/python3.7/site-packages/stellargraph/data/explorer.py in <module>
     29 import warnings
     30 from collections import defaultdict, deque
---> 31 from scipy import stats
     32 from scipy.special import softmax
     33 

~/anaconda3/envs/TF2/lib/python3.7/site-packages/scipy/stats/__init__.py in <module>
    386 
    387 """
--> 388 from .stats import *
    389 from .distributions import *
    390 from .morestats import *

~/anaconda3/envs/TF2/lib/python3.7/site-packages/scipy/stats/stats.py in <module>
    174 from scipy.spatial.distance import cdist
    175 from scipy.ndimage import measurements
--> 176 from scipy._lib._util import (_lazywhere, check_random_state, MapWrapper,
    177                               rng_integers)
    178 import scipy.special as special

ImportError: cannot import name 'rng_integers' from 'scipy._lib._util' (/home/zihan/anaconda3/envs/TF2/lib/python3.7/site-packages/scipy/_lib/_util.py)
```

**Comment by huonw** [comment]
That's very strange! It looks like a scipy submodule is failing to import another scipy internal submodule. This suggests to me that the problem isn't directly related to StellarGraph, but is instead a corrupted installation of scipy... but it's interesting that reinstalling didn't help. StellarGraph runs successfully on my macOS machine with scipy 1.4.1 (installed with pip), and on Ubuntu CI with scipy 1.5.2 (pip) and scipy 1.5.0 (conda).

To help narrow down potential issues here, I've opened https://github.com/stellargraph/stellargraph/pull/1785, which pins the version of scipy to 1.4.1 exactly, and thus will run through the full test suite and all of the demo notebooks with that version. Once CI has finished on it, hopefully we'll have more insight into the issue.

**Comment by huonw** [comment]
It looks like the Ubuntu pip (on all Python versions) and Conda (Python 3.8)  testing passed, using Scipy 1.4.1. For example, line 748 of https://github.com/stellargraph/stellargraph/pull/1785/checks?check_run_id=964620995 is `scipy:                    1.4.1-py37h0b6359f_0`. (Windows failed, but I think that was due to a different problem.)

As an additional check, I've switched Conda testing to use Python 3.7 in that PR. [That also passed](https://github.com/stellargraph/stellargraph/pull/1785/checks?check_run_id=964659996).

Thus, my remaining thought is that somehow scipy is failing to install properly in your environment. Could you try just importing it in without going via StellarGraph? For example: `python -c 'from scipy import stats'`, using the appropriate `python` command/binary for your Conda Python environment.

**Comment by ZihanChen1995** [comment]
Hi Huon! Thank you a lot for your help and reply. I just tried the way you said: Importing `scipy` directly and trying different functions. All of them work normally. 

Hence I'm thinking maybe it's because the `scripy` may not be installed correctly when I installed StellarGraph. Hence I reinstall `scipy` as well as `Stellargraph` through `pip` again instead of `conda`. It works perfectly now.

Thank you so much for your time, Huonw! It's solved now.
(To double-check, I created a new virtual environment via `conda` and installed `Stellargraph` before any other packages. It also works perfectly.)
